### PR TITLE
ztp: Update Readme and examples

### PR DESCRIPTION
**Pull request by serngawy**
Signed-off-by: melserngawy <melserng@redhat.com>

### CODE REVIEW

File: ztp/policygenerator-kustomize-plugin/README.md [hunk 1]
```diff
@@ -1,496 +1,93 @@
...
-        ├── site-du-sno-1-perfprofile-policy.yaml
-        ├── site-du-sno-1-policies-placementbinding.yaml
-        ├── site-du-sno-1-policies-placementrule.yaml
-        ├── site-du-sno-1-sriov-nnp-fh-policy.yaml
-        ├── site-du-sno-1-sriov-nnp-mh-policy.yaml
-        ├── site-du-sno-1-sriov-nw-fh-policy.yaml
-        └── site-du-sno-1-sriov-nw-mh-policy.yaml
+## Policy generator kustomize plugin
+
+The policy generator kustomize plugin consume the [policygenerator](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/policygenerator) lib and [PolicyGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml) as a kustomize plugin. Kustomization.yaml is an example how to use the policygenerator plugin as below
```
Comment (lack) [inline ztp/policygenerator-kustomize-plugin/README.md:3]:
```suggestion
The policy generator kustomize plugin consumes the [policygenerator](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/policygenerator) library and [PolicyGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml) as a kustomize plugin. Kustomization.yaml is an example of how to use the policygenerator plugin.
```

File: ztp/policygenerator-kustomize-plugin/README.md [hunk 2]
```diff
@@ -1,496 +1,93 @@
...
+```
+
+
+## Build and execute
+- Run the following command to build policygenerator binary and create the kustomize plugin directory
+```
+    $ make build
+```
+
+After the build success `kustomize/` directory should be created with following tree
```
Comment (lack) [inline ztp/policygenerator-kustomize-plugin/README.md:19]:
```suggestion
After the build succeeds, the `kustomize/` directory should be created with following structure:
```

File: ztp/policygenerator-kustomize-plugin/README.md [hunk 3]
```diff
@@ -1,496 +1,93 @@
...
+```
+
+Note: The source-crs directory contain the CRs (Custom Resources) that will be used to construct the ACM policies.
+
+- Run the following command to execute kustomization.yaml
+```
+    $ make test
+```
+
+- Run the following command to dump the kusomization output to files under `out/` directory
```
Comment (lack) [inline ztp/policygenerator-kustomize-plugin/README.md:90]:
```suggestion
- Run the following command to dump the kustomization output to files under the `out/` directory
```

File: ztp/policygenerator-kustomize-plugin/README.md [hunk 4]
```diff
@@ -1,496 +1,93 @@
...
+generators:
+- testPolicyGenTemplate/common-ranGen.yaml
+- testPolicyGenTemplate/group-du-ranGen.yaml
+- testPolicyGenTemplate/group-du-sno-ranGen.yaml
+- testPolicyGenTemplate/site-du-sno-1-ranGen.yaml
+```
+
+
+## Build and execute
+- Run the following command to build policygenerator binary and create the kustomize plugin directory
```
Comment (lack) [inline ztp/policygenerator-kustomize-plugin/README.md:14]:
```suggestion
- Run the following command to build the policygenerator binary and create the kustomize plugin directory structure:
```

File: ztp/policygenerator/README.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
+## PolicyGen (Policy Generator)
+PolicyGen lib is used to facilitate creating ACM policies based on given CRs (custom resources) and [PolicyGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml).
```
Comment (lack) [inline ztp/policygenerator/README.md:2]:
```suggestion
The PolicyGen library is used to facilitate creating ACM policies based on a set of provided source CRs (custom resources) and [PolicyGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml) CRs which describe how to customize those source CRs.
```

File: ztp/policygenerator/README.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+List of the CRs that ztp RAN solution provide to deploy ACM policies are under the [ztp/source-crs](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/source-crs). PolicyGenTemplate construct the ACM policy offering the following;
```
Comment (lack) [inline ztp/policygenerator/README.md:3]:
```suggestion
The full list of the CRs that ztp RAN solution provide to deploy ACM policies are under the [ztp/source-crs](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/source-crs). PolicyGenTemplate constructs the ACM policies by offering the following customization mechanisms:
```

And should we mention the source CRs are also embedded in our container image?

File: ztp/policygenerator/README.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+  1. Overlay; The given CRs that will be constructed into ACM policy could be override its attributes.
```
Comment (lack) [inline ztp/policygenerator/README.md:4]:
```suggestion
  1. Overlay: The given CRs that will be constructed into ACM policy may have some or all of their contents replaced by values specified in the PolicyGenTemplate.
```

File: ztp/policygenerator/README.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
+## PolicyGen (Policy Generator)
+PolicyGen lib is used to facilitate creating ACM policies based on given CRs (custom resources) and [PolicyGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml).
+List of the CRs that ztp RAN solution provide to deploy ACM policies are under the [ztp/source-crs](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/source-crs). PolicyGenTemplate construct the ACM policy offering the following;
+  1. Overlay; The given CRs that will be constructed into ACM policy could be override its attributes.  
+  1. Grouping; Policies defined in the PolicyGenTemplate will be created under the same namespace AND share the PlacmentRules and PlacementBinding.
```
Comment (lack) [inline ztp/policygenerator/README.md:5]:
```suggestion
  1. Grouping: Policies defined in the PolicyGenTemplate will be created under the same namespace and share the same PlacmentRules and PlacementBinding.
```

File: ztp/policygenerator/README.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+- Example 1: considering the PolicyGenTemplate below to create ACM policy constructing [ConsoleOperatorDisable.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/source-crs/ConsoleOperatorDisable.yaml) and [ClusterLogging.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/source-crs/ClusterLogging.yaml).
```
Comment (lack) [inline ztp/policygenerator/README.md:7]:
```suggestion
- Example 1: Consider the PolicyGenTemplate below to create ACM policies for both [ConsoleOperatorDisable.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/source-crs/ConsoleOperatorDisable.yaml) and [ClusterLogging.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/source-crs/ClusterLogging.yaml).
```

File: ztp/policygenerator/README.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+                curator:
+                  schedule: 30 3 * * *
+                type: curator
+              managementState: Managed
+        remediationAction: enforce
+        severity: low
+  remediationAction: enforce
+```
+
+The placement binding and rules of the generated policies will be as below.
```
Comment (lack) [inline ztp/policygenerator/README.md:125]:
```suggestion
The placement binding and rules of the generated policies will be:
```

File: ztp/policygenerator/README.md [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+        curation:
+          curator:
+            schedule: "30 3 * * *"
+        collection:
+          logs:
+            type: "fluentd"
+            fluentd: {}
+```
+
+The generated policies will be as below;
```
Comment (lack) [inline ztp/policygenerator/README.md:33]:
```suggestion
The generated policies will be:
```

File: ztp/policygenerator/README.md [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+    - key: group-du-sno
+      operator: In
+      values:
+      - ""
+```
+## Build and execute
+- Requirement
+  - golang is installed
+
+- Run the following command to build policygenerator binary
```
Comment (lack) [inline ztp/policygenerator/README.md:162]:
```suggestion
- Run the following command to build the policygenerator binary:
```

File: ztp/policygenerator/README.md [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+```
+    $ make build
+```
+
+- Run the following command to execute the unit test
```
Comment (lack) [inline ztp/policygenerator/README.md:167]:
```suggestion
- Run the following command to execute the unit tests:
```

File: ztp/policygenerator/README.md [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+```
+    $ make test
+```
+
+- Run the following command to execute policygenerator binary with a PolicyGenTemplate example
+```
+    $ ./policygenerator  -sourcePath ../source-crs ../ran-crd/policy-gen-template-ex.yaml
+```  
+
+- Run the following command to see help
```
Comment (lack) [inline ztp/policygenerator/README.md:177]:
```suggestion
- Run the following command to see the command's help text:
```

File: ztp/policygenerator/README.md [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+    	Directory to write the genrated policies (default "__unset_value__")
+  -pgtPath string
+    	Directory where policyGenTemp files exist (default "__unset_value__")
+  -sourcePath string
+    	Directory where source-crs files exist (default "source-crs")
+  -wrapInPolicy
+    	Wrap the CRs in acm Policy (default true)
+```
+
+- For using policygenerator library as kustomize plugin check the [policy-generator-kustomize-plugin](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/policygenerator-kustomize-plugin/README.md)
```
Comment (lack) [inline ztp/policygenerator/README.md:191]:
```suggestion
- For using policygenerator library as kustomize plugin, see [policy-generator-kustomize-plugin](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/policygenerator-kustomize-plugin/README.md).
```

**Review by lack — CHANGES_REQUESTED** [review]

File: ztp/gitops-subscriptions/argocd/README.md [hunk 16]
```diff
@@ -121,27 +121,51 @@ Status:
...
 The expected set of policy wrapped CRs should be displayed.
 
-1. Did the PolicyGenTemplate get syncronized to the hub cluster?  
+If the Policies are not synchronized follow the steps below for failure to synchronize.
+
+```
+    $ oc describe -n openshift-gitops application policies 
+```
+
+Check for `Status: Conditions:` it will show error logs ex; setting invalid `sourceFile->fileName:` will generate error.
```
[the remark this answers is not in the document]
Comment (serngawy) [reply ztp/gitops-subscriptions/argocd/README.md]: done

File: ztp/gitops-subscriptions/argocd/README.md [hunk 17]
```diff
@@ -121,27 +121,51 @@ Status:
  
 
 ### Validate generation of configuration policy CRs
-Policy CRs are generated in same namespace as the PolicyGenTemplate from which they were created. This same troubleshooting flow applies to all policy CRs generated from PolicyGenTemplates regardless of whether they are common, group or site based.  
+
+Policy CRs are generated in same namespace as the PolicyGenTemplate from which they were created. The same troubleshooting flow applies to all policy CRs generated from PolicyGenTemplates regardless of whether they are ztp-common, ztp-group or ztp-site based.
```
[the remark this answers is not in the document]
Comment (serngawy) [reply ztp/gitops-subscriptions/argocd/README.md]: done

File: ztp/gitops-subscriptions/argocd/README.md [hunk 18]
```diff
@@ -121,27 +121,51 @@ Status:
...
+
+Policy CRs are generated in same namespace as the PolicyGenTemplate from which they were created. The same troubleshooting flow applies to all policy CRs generated from PolicyGenTemplates regardless of whether they are ztp-common, ztp-group or ztp-site based.  
 ```
     $ export NS=<namespace>
     $ oc get policy -n $NS
 ```
 The expected set of policy wrapped CRs should be displayed.
 
-1. Did the PolicyGenTemplate get syncronized to the hub cluster?  
+If the Policies are not synchronized follow the steps below for failure to synchronize.
```
[the remark this answers is not in the document]
Comment (serngawy) [reply ztp/gitops-subscriptions/argocd/README.md]: done

File: ztp/gitops-subscriptions/argocd/README.md [hunk 19]
```diff
@@ -121,27 +121,51 @@ Status:
...
+Status:
+  Conditions:
+    Last Transition Time:  2021-11-26T17:21:39Z
+    Message:               rpc error: code = Unknown desc = `kustomize build /tmp/https___git.com/ran-sites/policies/ --enable-alpha-plugins` failed exit status 1: 2021/11/26 17:21:40 Error could not find test.yaml under source-crs/: no such file or directory
+Error: failure in plugin configured via /tmp/kust-plugin-config-52463179; exit status 1: exit status 1
+    Type:  ComparisonError
 ```
 
-If the PolicyGenTemplate is not synchronized follow the steps above for failure to synchronize the SiteConfig CR to the hub.
+Check for `Status: Sync:` if there are log errors the `Sync: Status:` could be as below `Unknown` / `Error`.
```
[the remark this answers is not in the document]
Comment (serngawy) [reply ztp/gitops-subscriptions/argocd/README.md]: Just to show the Sync->status is error or unknown as the Sync->condition show the logs

**Review by lack — APPROVED** [review]

**Pull request merged**